cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -528,56 +531,55 @@ txOutToJsonValue era (TxOut addr val dat refScript) =
     ShelleyEra -> object ["address" .= addr, "value" .= val]
     AllegraEra -> object ["address" .= addr, "value" .= val]
     MaryEra -> object ["address" .= addr, "value" .= val]
-    AlonzoEra -> object $ catMaybes
-                   [ Just ("address" .= addr)
-                   , Just ("value" .= val)
+    AlonzoEra -> object
+                   [ "address" .= addr
+                   , "value" .= val
                    , datHashJsonVal dat
-                   , datJsonVal dat
+                   , "datum" .= datJsonVal dat
                    ]
     BabbageEra ->
-      object $ catMaybes
-        [ Just ("address" .= addr)
-        , Just ("value" .= val)
+      object
+        [ "address" .= addr
+        , "value" .= val
         , datHashJsonVal dat
-        , datJsonVal dat
-        , inlineDatumJsonVal dat
-        , refScriptJsonVal refScript
+        , "datum" .= datJsonVal dat
+        , "inlineDatum" .= inlineDatumJsonVal dat
+        , "referenceScript" .= refScriptJsonVal refScript
         ]
  where
-   datHashJsonVal :: TxOutDatum ctx era -> Maybe Aeson.Pair
+   datHashJsonVal :: TxOutDatum ctx era -> Aeson.Pair
    datHashJsonVal d =
      case d of
-       TxOutDatumNone -> Nothing
+       TxOutDatumNone ->
+         "datumhash" .= Aeson.Null
```

Comment:
Behavioural change where absence maps to `field = null` rather than absence of field.  Is this intended?

Reply:
Yeah the JSON instances were a pain to get round tripping. I opted for `Null` instead to make it simpler. Even then this needs to be revisited and refactored because its too complicated.